weatherAplication/Modules/ShowWeather/ViewModel/ShowWeatherViewModel.swift

```diff
@@ -28,13 +30,20 @@ class ShowWeatherViewModel: ShowWeatherViewModelContract {
     }
 
     func getWeather() {
-        apiClient.searchWeather(latitude: latitude, longitude: longitude, completion: { [weak self] weatherInfo, date -> Void in
-            self?.dateString = date
-            self?.weatherDataArray = weatherInfo
-            self?.delegate?.weatherLoaded(weatherInfo: weatherInfo, date: date)
-            }, failure: { weatherError in
-                self.delegate?.showAlert(description: weatherError.localizedDescription)
-            })
+        cancellable = apiClient.searchWeather(latitude: latitude, longitude: longitude)
+            .sink(
+                receiveCompletion: { completion in
+                    switch completion {
+                    case.failure(let error):
+                        self.delegate?.showAlert(description: error.localizedDescription)
+                    case.finished:
+                        print("finished")
+                    }
+                }, receiveValue: {[weak self] weatherInfo, date in
+                    self?.dateString = date
+                    self?.weatherDataArray = weatherInfo
+                    self?.delegate?.weatherLoaded(weatherInfo: weatherInfo, date: date)
+                })
```

Comment:
nie potrzebujesz sprawdzac co sie stanie dla receiveCompletion, to tak naprawde niekoniecznie musi byc blad, wystarczy ze zrobisz sink { [obsluga dla receiveValue] }